IHP/ModelSupport.hs

```diff
@@ -935,3 +935,41 @@ withTableReadTracker trackedSection = do
     let ?modelContext = oldModelContext { trackTableReadCallback }
     let ?touchedTables = touchedTablesVar
     trackedSection
+
+
+-- | Shorthand filter function
+--
+-- In IHP code bases you often write filter functions such as these:
+--
+-- > getUserPosts user posts =
+-- >     filter (\p -> p.userId == user.id) posts
+--
+-- This can be written in a shorter way using 'onlyWhere':
+--
+-- > getUserPosts user posts =
+-- >     posts |> onlyWhere #userId user.id
+--
+-- Because the @userId@ field is an Id, we can use 'onlyWhereReferences' to make it even shorter:
+--
+-- > getUserPosts user posts =
+-- >     posts |> onlyWhereReferences #userId user
+--
+-- If the Id field is nullable, we need to use 'onlyWhereReferencesMaybe':
+--
+-- > getUserTasks user tasks =
+-- >     tasks |> onlyWhereReferences #optionalUserId user
+--
+onlyWhere :: forall record fieldName value. (KnownSymbol fieldName, HasField fieldName record value, Eq value) => Proxy fieldName -> value -> [record] -> [record]
+onlyWhere field value records = filter (\record -> get field record == value) records
+
+-- | Shorthand filter function for Id fields
+--
+-- See 'onlyWhere' for more details.
```

Comment:
I think worth adding docs also to this one, as for me it's not clear when/ how to use

Reply:
Good point 👍  just pushed an update

Reply:
Thanks, looks good